### PR TITLE
BE-011 Hanna Kalashnyk User-coupon-favourite

### PR DESCRIPTION
**Pull request by Hanna-Kalashnyk**

### CODE REVIEW

File: src/main/java/com/exadel/discount/DiscountApplication.java [hunk 1]
```diff
@@ -1,7 +1,12 @@
 package com.exadel.discount;
 
+import org.hibernate.validator.spi.messageinterpolation.LocaleResolver;
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.Bean;
+import org.springframework.web.servlet.i18n.SessionLocaleResolver;
+
```
Comment (rudik-byte) [inline src/main/java/com/exadel/discount/DiscountApplication.java]: We really need this imports ?

File: src/main/java/com/exadel/discount/config/SpringFoxConfig.java [hunk 2]
```diff
@@ -1,28 +0,0 @@
...
-                .paths(PathSelectors.any())
-                .build()
-                .apiInfo(new ApiInfoBuilder()
-                        .title("Discount API")
-                        .description("API for the Discount Application")
-                        .version("1.0")
-                        .build()
-                );
-    }
-}
```
Comment (el-dub) [inline src/main/java/com/exadel/discount/config/SpringFoxConfig.java]: Why did you delete this file? Maybe it's a mistake in merging

**Review by el-dub — CHANGES_REQUESTED** [review]

File: src/main/java/com/exadel/discount/dto/coupon/CouponDto.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import javax.persistence.JoinColumn;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.Null;
+import javax.validation.constraints.Size;
+import java.time.LocalDateTime;
+import java.util.UUID;
+
+@Data
+public class CouponDto {
+    @Null(groups = Create.class, message = "Coupon id should be null")
```
Comment (rudik-byte) [inline src/main/java/com/exadel/discount/dto/coupon/CouponDto.java:19]: maybe "not null"?

File: src/main/java/com/exadel/discount/config/SpringFoxConfig.java (hunk 2 above)
Comment (Hanna-Kalashnyk) [reply src/main/java/com/exadel/discount/config/SpringFoxConfig.java]: Is it back now?

**Comment by Hanna-Kalashnyk** [comment]
Hi!
I think, @Null, because During creative POST request  new coupon has not ID
yet.
Is the Discount class yours? I still need such block there?


*@OneToMany* <https://github.com/OneToMany>(mappedBy = "discounts", fetch =
FetchType.LAZY, cascade = CascadeType.REMOVE)

private Set coupons;

*@OneToMany* <https://github.com/OneToMany>(mappedBy = "discounts", fetch =
FetchType.LAZY, cascade = CascadeType.REMOVE)

private Set favorites;


Could U aslo watch my PR , please ...
I have added parts with  swagger,  slt4j  and exceptions  in it :
https://github.com/SergeyKoval/2021_june_team4_be/pull/24/files
пт, 18 июн. 2021 г. в 15:49, Anton Rudkovskyi ***@***.***>:

**Comment by Hanna-Kalashnyk** [comment]
Sorry, I need this one:

*@OneToMany* <https://github.com/OneToMany>(mappedBy = "discount", fetch =
FetchType.LAZY, cascade = CascadeType.REMOVE)

private Set coupons;

*@OneToMany* <https://github.com/OneToMany>(mappedBy = "discount", fetch =
FetchType.LAZY, cascade = CascadeType.REMOVE)

private Set favorites;


Or is this still to Nicolay?

вт, 22 июн. 2021 г. в 15:02, Hanna Kalashnyk ***@***.***>:
